src/samples/__tests__/e2e.test.ts

```diff
@@ -145,27 +145,20 @@ class PollingInboundTransporter implements InboundTransporter {
   async registerAgency(agent: Agent) {
     const agencyUrl = agent.getAgencyUrl() || '';
     const agencyInvitationUrl = await get(`${agencyUrl}/invitation`);
-    const agencyInvitation = decodeInvitationFromUrl(agencyInvitationUrl);
-    const agentConnectionAtAgency = await agent.acceptInvitation(agencyInvitation);
-
-    this.pollMessages(agent, agencyUrl, agentConnectionAtAgency.verkey);
-
-    if (!agentConnectionAtAgency) {
-      throw new Error('Connection not found!');
-    }
-    await agentConnectionAtAgency.isConnected();
-    console.log('agentConnectionAtAgency\n', agentConnectionAtAgency);
-
     const { verkey: agencyVerkey } = JSON.parse(await get(`${agencyUrl}/`));
-    agent.establishInbound(agencyVerkey, agentConnectionAtAgency);
+    await agent.provision({ verkey: agencyVerkey, invitationUrl: agencyInvitationUrl });
+    this.pollDownloadMessages(agent);
   }
 
-  pollMessages(agent: Agent, agencyUrl: string, verkey: Verkey) {
+  pollDownloadMessages(agent: Agent) {
```

Comment:
I am going to add this comment realizing I may not be fully understanding what is intended (since Im just getting back into aries).  Would it be better to keep pollMessages and create a new method pollDownloadMessages?

Reply:
Thanks for your comment @matt-raffel-kiva It made me think twice, but the previous implementation of `pollMessages` was my custom way how to achieve the required behavior unrelated to any Aries RFCS, so throwing it away is the right choice here. This newer implementation does the same but relies more on Aries RFCS mentioned in #40